server/src/main/scala/org/http4s/server/middleware/authentication/DigestAuth.scala

```diff
@@ -131,6 +131,10 @@ object DigestAuth {
       challenged(runChallenge)
     }
 
+  @deprecated(
+    "Uses a side-effecting NonceKeeper. Use challenge(String, AuthStore, Blocker, Duration, Int, Int).",
+    "0.22.13",
+  )
   def challenge[F[_], A](
```

Comment:
I'm not sure whether this signature is still valid, but I thought `NonceKeeper` was considered dirty. 

/cc @blast-hardcheese

Reply:
It's impossible to be called publicly, since `NonceKeeper` is package private -- I had a `@deprecated` on it but it must have gotten lost. Thanks for putting it back